asp.net.mvc_lesson2/asp.net.mvc_lesson2/asp.net.mvc_lesson2/ThreadPul/PulThread.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading;
+
+
+namespace asp.net.mvc_lesson2
+{
+   ///  <summary> запись и чтение ряда фибонначи </summary>
+    public class PulThread
+    {
+        private static Mutex _mutex = new Mutex(); 
+
+        public ConcurrentDictionary<int, DateTime> ListFobonacci = new ConcurrentDictionary<int , DateTime>();
+
+
+
+       
+
+        public int n2 { get; set; }
+
+
+
+        internal DateTime Time1 { get; set; }
+        internal DateTime Time2 { get; set; }
+        internal int N { get; set; } = 0;
+
+
+
+
+        ///<returns> добавляет  в словарь ListFobonacci числовой ряд фибонначи (колмчество элементов n) как ключ и как значения время вычисления ( задержка milliSeconds )  </returns> 
+        internal void ListFobonacciAdd( int n, int milliSeconds) 
+        {
+
+
+            int _n1 = 0; // n-1
+            int _n = 1; // n
+
+            var Count = ListFobonacci.GetEnumerator();
```

Comment:
```suggestion
            using var Count = ListFobonacci.GetEnumerator();
```
`IEnumerator` может требовать освобождения ресурсов (он реализует `IDisposable`, следовательно, требует конструкции `using(...) { ... }`).